### PR TITLE
Add ability to cancel previous matches upon new match creation

### PR DESCRIPTION
**Pull request by noah-solomon**
## Overview

Every week when new matches are created, we have to manually cancel old matches by changing their `status` field on Postgres. This PR enables the cancelation of old matches in a much easier manner, by simply setting `cancel_previous=true` in the request body for creating new matches.

Also: refactored some messy code/naming and added new endpoint to see all matches

For updated API spec, see the SP22 API Specification in Notion.

## Changes Made

- Added check for `cancel_previous` flag in `POST /api/matches/` and `POST /api/matches/multiple/`
- Moved match view for specific user into `match` directory for consistency
- Added new endpoint `GET /api/matches/all/` to allow `superuser` to see all matches ever created
- Updated API Specification in Notion
- Changed naming of views for clarity:
  - `AllMatchesView` is now `UserMatchesView` because it's all of the matches for one user
  - `MatchesView` is now `MyMatchesView` because it's the matches for "me" i.e. authenticated user
  
## Test Coverage

Tested in Postman

### CODE REVIEW

File: src/match/controllers/create_match_controller.py [hunk 1]
```diff
@@ -44,6 +46,15 @@ def process(self):
         possible_match = Match.objects.filter(
             user_1=user_1, user_2=user_2, status=match_status.CREATED
         )
+        if cancel_previous:
+            user_1_previous_matches = Match.objects.filter(
+                Q(user_1=user_1) | Q(user_2=user_1)
+            )
+            user_2_previous_matches = Match.objects.filter(
+                Q(user_1=user_2) | Q(user_2=user_2)
+            )
```
Comment (alanna-zhou) [inline src/match/controllers/create_match_controller.py:55]: nice :)

**Review by alanna-zhou — APPROVED** [review]
amazingg~~~

**Review by katesliang — APPROVED** [review]
looks valid!! Q is so cool!!!!

File: src/person/controllers/mass_message_controller.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def process(self):
         if sandbox:
             # no messages were sent
             return success_response(
-                data=f"{notifs_enabled_count} users have push notifications enabled",
+                data=f"{notifs_enabled_count} users have push notifications enabled, no notfications sent (sandbox mode)",
```
Comment (chalo2000) [inline src/person/controllers/mass_message_controller.py:61]: this doesn't look related to this PR, what is 'sandbox mode'?

**Review by chalo2000 — APPROVED** [review]
LGTM just confused about some notification string change (leave it, I'm just not sure if it relates to this since you didn't mention it in your PR report)

File: src/person/controllers/mass_message_controller.py (hunk 2 above)
Comment (noah-solomon) [reply src/person/controllers/mass_message_controller.py:61]: ah this is a smol change, sandbox mode allows us to see how many people have notifs enabled without actually sending a notif to everyone, I recently added it to our API spec
